src/ChessBoard/MovePieces.tsx

```diff
@@ -1,134 +1,493 @@
-        };
+        }
     }
 
-    onMouseDown = (event: React.MouseEvent<HTMLCanvasElement>) => {
-        const mousePositions = this.getMousePositions(event)
-        const index = this.getIndexAtPosition(mousePositions.x, mousePositions.y);
+    function onMouseDown (event: React.MouseEvent<HTMLCanvasElement>) {
+        const mousePositions = getMousePositions(event)
+        const index = getIndexAtPosition(mousePositions.x, mousePositions.y);
         if (index !== -1) {
-            this.setState({ isDragging: true, draggingIndex: index });
-            window.addEventListener("pointermove", this.onMouseMove);
-            window.addEventListener("pointerup", this.onMouseUp);
-            this.setState({ mousePosition: { x: mousePositions.x, y: mousePositions.y } });
+            draggingIndex = index
+            mousePosition = {x: mousePositions.x, y: mousePositions.y};
         }
     }
 
-    onMouseUp = () => {
-        this.setState({ isDragging: false, draggingIndex: -1 });
-        window.removeEventListener("pointermove", this.onMouseMove);
-        window.removeEventListener("pointerup", this.onMouseUp);
+    const onMouseUp = () => {
+        adjustPiecePosition()
+        draggingIndex = -1
     }
 
-    handleMouseMove = (event: MouseEvent) => {
-        const mousePositions = this.getMousePositions(event)
+    const handleMouseMove = (event: MouseEvent) => {
+        const mousePositions = getMousePositions(event)
         let x = mousePositions.x
         let y = mousePositions.y
         //Can't move pieces outside of canvas
-        if (x >= 550) x = 550
+        const border = canvasSize - imageSize
+        if (x >= border) x = border
         if (x <= 0) x = 0
         if (y <= 0) y = 0
-        if (y >= 550) y = 550
-        this.setState({ mousePosition: { x: x, y: y } });
+        if (y >= border) y = border
+        mousePosition = { x: x, y: y };
     }
 
-    onMouseMove = (event: MouseEvent) => {
-        const mousePositions = this.getMousePositions(event)
-        let x = mousePositions.x
-        let y = mousePositions.y
-        // Update position of the piece to be in the middle of the square
-        for (let xPosition = 0; xPosition < 600; xPosition+=75){
-            for (let yPosition = 0; yPosition < 600; yPosition+=75){
-                if ((this.state.mousePosition.x >= xPosition && this.state.mousePosition.x <= xPosition + 75) &&
-                    (this.state.mousePosition.y >= yPosition && this.state.mousePosition.y <= yPosition + 75)){
-                        x = xPosition + 12.5
-                        y = yPosition + 12.5
+    const onMouseMove = (event: MouseEvent) => {
+        mousePosition = getMousePositions(event);
+    }
+
+    const adjustPiecePosition = () => {
+        let nextSquare = squareSize;
+        const pieceImageShift = (squareSize - imageSize) / 2;
+        let x = mousePosition.x
+        let y = mousePosition.y
+        const i = draggingIndex
```

Comment:
Do we really need new variable for draggingIndex? `draggingIndex` name is better self explaining name then` i`